corehq/form_processor/submission_post.py

```diff
@@ -541,3 +541,20 @@ def unfinished_submission(instance):
     tracker = SubmissionProcessTracker(unfinished_submission_stub)
     yield tracker
     tracker.submission_fully_processed()
+
+
+@contextlib.contextmanager
+def unfinished_archive(instance, user_id, archive):
```

Comment:
since this is unrelated to submissions consider moving to a different place (also fine to move the other context manager)

Reply:
Moved to a separate file: https://github.com/dimagi/commcare-hq/pull/22470/commits/dae93ea4d3ff1e0276e861accad83db7623eebe2